lib/parslet/context.rb

```diff
@@ -32,4 +24,31 @@ def initialize(bindings)
       instance_variable_set("@#{key}", value)
     end
   end
+
+  # following methods are needed and were provided by blankslate
```

Comment:
The purpose of BlankSlate is to have a clean context for transformation code to run in. Clean in this sense means: Free from all the monkey-patching that goes on in the ruby world. But to usefully run that code, some methods still need to work; uncovering more methods should be easy to do. 

I've had to decline several attempts at replacing this construct with BasicObject, so I might really sound (and actually be) impatient at the topic. The misconception most people have is that BlankSlate is basically BasicObject, but badly done. That is simply not true. BasicObject (if we remain on that higher plane for a second) is a lot less than BlankSlate. 

So to replace BlankSlate, we basically need to emulate a lot of it's functionality. I didn't mean to rub you wrong, just calling things out as they are: If we just satisfy tests here, we'll not achieve any of the above goals. So the tests are really incomplete. That's another fact. 

I see three ways forward here: 

a) Make blankslate work in opal
b) Emulate BlankSlate for parslet by writing our own blankslate, based on BasicObject
c) Loosen the goals a bit and use plain object. 

And yeah, I've obviously haven't got the time on my hands to help make a good PR. Believe me, if someone really renders BlankSlate obsolete (maybe even by method (c), though that would take some convincing), I'll press the green merge button gladly. But reading the upteenth attempt is really getting tiring. 

So that is the bad vibe you're getting. You're correct with that. Nothing personal though; and I'll read everything you do with a benevolent eye and merge once the quality is there.